controllers/userController.js

```diff
@@ -0,0 +1,47 @@
+const User = require("../models/User")
+require("dotenv").config();
+const { signToken } = require("../auth")
+const StatusCodes = require('http-status-codes')
+
+
+module.exports = {
+    signUp: (req , res) => {
+        // check wether user exists before creating one
+        User.findOne({email: req.body.email}).then(user => {
```

Comment:
you could destructure email from req.body;
```suggestion
const { email, role, password } = req.body
        User.findOne({email: email}).then(user => {
```